### PR TITLE
Feat/fetch image data

### PR DESCRIPTION
**Pull request by sdmisra**
- change images to use API instead of test data - need to make dynamic w form
- header is now a link back to homepage
- rovers endpoint is slow, we will need to add loading screen in later iteration

### CODE REVIEW

**Review by Zertroz — APPROVED** [review]
Images work, next step is form

**Review by kirch1 — COMMENTED** [review]
Looks good, will need a fancy loading screen because this API is pretty slow :(

**Review by bea-ordonez — COMMENTED** [review]
Let there be images!

File: cypress/e2e/dashboard.cy.js [hunk 1]
```diff
@@ -4,25 +4,33 @@ import roversFixture from "../fixtures/roversFixture";
...
+      .should('be.visible')
+      .contains('Mars Chronicles');
+
   });
 
   it('user should see all rover images and they should have alt text', () => {
+
     cy.get('[href="/Curiosity"] > div > .rover-icon-img')
      .should('exist').should('be.visible')
      .should('have.attr', 'alt');
```
Comment (hfaerber) [inline cypress/e2e/dashboard.cy.js:29]: I love seeing the testing for alt text here!

File: cypress/e2e/roverPage.cy.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// This test will be concerned with the transition from the main page, as well as the rendering of multiple components that display more detailed information about a chosen rover.
+
+// User flows to test will include: details, model, form, and gallery rendering and specific values within each component. 
+// Sad path: Error handling for search refining that returns no results, EH for network errors when attempting to access rover details or images, too.
```
Comment (hfaerber) [inline cypress/e2e/roverPage.cy.js:4]: Love the attention to sad paths here

File: cypress/e2e/roverPage.cy.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  })
+
+  it('When a user chooses a rover, the rover icons should disappear but the header should remain:', () => {
+
+    cy.get('h1')
+      .should('exist').should('be.visible')
+      .contains('Mars Chronicles');
+  
+    cy.get('[href="/Curiosity"] > div > .rover-icon-img')
+     .should('not.exist')
```
Comment (hfaerber) [inline cypress/e2e/roverPage.cy.js:31]: Love the use of 'not.exist'

File: src/App.js [hunk 4]
```diff
@@ -7,16 +7,20 @@ import RoverPage from './features/RoverPage/RoverPage';
...
 
 function App() {
   const dispatch = useDispatch()
-  
-  useEffect(async () => {
+  const fetchData = async () => {
     const roverData = await fetchRovers()
     dispatch(setRovers(roverData.rovers))
-  }, [])
+  }
```
Comment (hfaerber) [inline src/App.js:17]: Good use of async/await.   Be sure that everyone one in the group fully understands async/await - to the point that they could explain it to me if asked.   *Each group member should take ownership of doing some solo study themself to ensure they understand this - the documentation, a blog post, a youtube video, talking to their house mentor about it, etc.    But the group as a whole should continue to check in on those efforts and help support each other in their understanding as well.

File: src/features/Apicalls.js [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
   return rovers
 }
 
-export default fetchRovers;
\ No newline at end of file
+const fetchImages = async () => {
+  const response = await fetch('https://api.nasa.gov/mars-photos/api/v1/rovers/opportunity/photos?sol=1&api_key=vnSCs1JJ7ERtXeAqN6cajKwEh99pz5q6xueRhxMV')
+  const photos = await response.json()
+  return photos
+}
```
Comment (hfaerber) [inline src/features/Apicalls.js:11]: Consider how you might include code to handle if you get a bad response from the network request.

File: src/features/header/Header.js [hunk 6]
```diff
@@ -1,11 +1,14 @@
 import './Header.css'
+import { Link } from 'react-router-dom'
 
 function Header() {
   return(
     <header>
-      <h1 className='header-logo'>Mars Chronicles</h1>
+      <Link to="/">
+        <h1 className='header-logo'>Mars Chronicles</h1>
+      </Link>
```
Comment (hfaerber) [inline src/features/header/Header.js:9]: Love that you've made the header a a link to the homepage - good consideration of making the user experience intuitive

**Comment by hfaerber** [comment]
Nice work team!  Love to see the thinking ahead to a loading state as you notice the slow load time.

**Pull request merged**